genai-perf/genai_perf/profile_data_parser/llm_profile_data_parser.py

```diff
@@ -161,14 +184,47 @@ def _parse_requests(self, requests: dict) -> Metrics:
                     chunked_inter_token_latency.append(round((t2 - t1) / num_token))
                 chunked_inter_token_latencies.append(chunked_inter_token_latency)
 
+                # (per-session) calculate llm metrics
+                if session_id:
+                    session_metric = self._session_metrics[session_id]
+                    session_metric["request_latencies"].append(req_latency_ns)
+                    session_metric["time_to_first_tokens"].append(ttft)
+                    if len(res_timestamps) > 1:
+                        session_metric["time_to_second_tokens"].append(ttst)
+                    if total_output_token > 1:
+                        session_metric["inter_token_latencies"].append(
+                            inter_token_latency
+                        )
+                    session_metric["output_token_throughputs_per_request"].append(
+                        total_output_token / req_latency_s
+                    )
+                    session_metric["input_sequence_lengths"].append(input_seq_len)
+                    session_metric["output_sequence_lengths"].append(total_output_token)
+                    # collect request and last response timestamps each session
+                    session_metric["req_timestamps"].append(req_timestamp)
+                    session_metric["last_res_timestamps"].append(res_timestamps[-1])
+
                 pbar.update(1)
 
         # request & output token throughput
         benchmark_duration = (max_res_timestamp - min_req_timestamp) / 1e9  # to seconds
         request_throughputs = [len(requests) / benchmark_duration]
         output_token_throughputs = [sum(output_sequence_lengths) / benchmark_duration]
 
-        llm_metric = LLMMetrics(
+        # (per-session) request & output token throughput
```

Comment:
Possible new method. Thoughts?

Reply:
For this one, we certainly could. I don't have much preference here.

Reply:
The method is long at this point.
Lets add a method here.

Reply:
Done